src/transport/SessionHandle.cpp

```diff
@@ -23,20 +23,30 @@ namespace chip {
 
 using namespace Transport;
 
-const PeerAddress * SessionHandle::GetPeerAddress(SessionManager * sessionManager) const
+Session * SessionHandle::operator->() const
 {
-    if (IsSecure())
+    if (mSession.Is<VariantSecureSession>())
     {
-        SecureSession * state = sessionManager->GetSecureSession(*this);
-        if (state == nullptr)
-        {
-            return nullptr;
-        }
-
-        return &state->GetPeerAddress();
+        return AsSecureSession();
+    }
+    else if (mSession.Is<VariantUnauthenticatedSession>())
+    {
+        return &AsUnauthenticatedSession().Get();
     }
 
-    return &GetUnauthenticatedSession()->GetPeerAddress();
+    VerifyOrDie(false);
+    return nullptr;
+}
+
+Transport::SecureSession * SessionHandle::AsSecureSession() const
+{
+    const VariantSecureSession & session = mSession.Get<VariantSecureSession>();
+    return session.mSessionManager.GetSecureSession(session.mLocalSessionId);
```

Comment:
So this can still return null, right?  But some of the callsites that use this (possibly indirectly now that it's hidden under `operator->` had their null-checks removed?  Why?